### PR TITLE
Solve command functionality bug & documentation bug

### LINKED ISSUE #286: [PE-D][Tester B] Incorrect add command
I enntered the same command from user guide but it says unknown command

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: JunLongling/ped#4

### LINKED ISSUE #288: [PE-D][Tester E] add command not working according to UG
in page 4 of the user guide, it states the following:
`add n/Samson s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id T1001 .

However, the application deems it as `Unknown command`

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: yyanyee/ped#4

### LINKED ISSUE #305: [PE-D][Tester B] Edit Command cannot edit the name of the person even though the student ID is different
I cannot edit the name of the first person to Alice even though the student ID is different as specify by the user guide

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: JunLongling/ped#10

### LINKED ISSUE #308: [PE-D][Tester D] addStu command states that name must be alphanumeric but names with numbers not accepted
When adding a student, names that include numbers are not accepted although the fail message suggests that alphanumeric names are accepted!

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: lim-jt/ped#2

### LINKED ISSUE #312: [PE-D][Tester B] When adding a person after the find command, it brings me back to the main list
After find Samson, I want to add Alice, but it brings me to the main list and my find result disappear

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: JunLongling/ped#11

### LINKED ISSUE #327: [PE-D][Tester D] Must-have user story not incorporated in product
The user story where "As a tutor I want to categorise students based on tutorial classes so that I can find out which students are in my class" is not implemented in the product as there is no way to filter the students based on class!

-------------
Labels: `type.DocumentationBug` `severity.High`
original: lim-jt/ped#8

### LINKED ISSUE #334: [PE-D][Tester B] Wrong Error command for addStu
At first, I have not added any tutorial.

But when i tried to add a student to a tutorial that has not been added

It should say You will need to add a tutorial before assign student to new tutorial

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#15

### LINKED ISSUE #335: [PE-D][Tester B] Adding a student with missing Student ID
From the developer guide, it say  Expected: Error message indicating that the student ID is missing.
but I got this

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#16

### LINKED ISSUE #336: [PE-D][Tester B] Adding a student with missing Name
Shows full command example instead of warning the user that Expected: Error message indicating that the student's name is missing as described in the developer guide.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#17

### LINKED ISSUE #337: [PE-D][Tester B] Adding a student with invalid student ID
From developer Guide,
Test case: addStu n/John Doe s/INVALID_ID t/T1001
Expected: Error message indicating that the student ID format is invalid.

Acutal result

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#18

### LINKED ISSUE #339: [PE-D][Tester B] Adding a student with invalid tutorial ID
Test case: addStu n/John Doe s/A1234567X t/INVALID_TUT
Expected: Error message indicating that the tutorial ID does not exist

got this instead

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#19

### LINKED ISSUE #341: [PE-D][Tester B] Unclear Developer Guide
When I tried the example given. I received this error

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#20

### LINKED ISSUE #342: [PE-D][Tester B] Editing a student with invalid index
Test case: edit 0 n/Jane Smith
Expected: Error message indicating that the student index provided is invalid.

gave full command instead of what is specify in the developer guide

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JunLongling/ped#21

### LINKED ISSUE #360: [PE-D][Tester A] Inconsistent mark command in user guide.
Inconsistent example and command name. The attend command seems to be the wrong command. 

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: Incogdino/ped#16

### LINKED ISSUE #364: [PE-D][Tester C] Add command provides wrong error message when provided name includes number
**Description:**  

In the UG, the `STUDENT_NAME` is described to be strictly containing alphabetic character and whitespaces.

**Steps to Reproduce:**
1. Use `addStu` to add any student whose name includes a number

**Expected Result:**  
An error message should be provided, explaining that we can't use numbers in the name.

**Actual Result:**  
The error message prompts us that alphanumeric characters are allowed (i.e. numbers are also allows), but prevents us from keying in numbers in names.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: apollo-tan/ped#11

### LINKED ISSUE #365: [PE-D][Tester C] Add command prevents us from adding a name with just whitespaces
**Description:**  

As per the User Guide, `STUDENT_NAME` can include "whitespaces" which means that having whitespaces alone should work.

However, in the application, specifying only whitespaces as names doesn't actually work.

**Suggestion:**
Perhaps you could fix the User Guide to explicitly mention that blank values are not allowed as name, much like what the error message is currently showing in the application.

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: apollo-tan/ped#12

### PR DESCRIPTION
**Pull request by SamsonChew**
- [x] student that has name with only whitespace is not allowed (documentation) (365)
- [x] student that has name with numeric character is not allowed (documentation) (364, 308)
- [x] make markAtt & deleteAtt command examples consistent (documentation) (360)
- [x] Change the DG manual testing session for add student command (documentation) (339, 337, 336, 335, 334)
- [x] Change user story in DG (327)
- [x] Explain find command potential scenarios on add & edit command in UG (documentation) (312)
- [x] Change section 3.1 Installation in UG for add command (288, 286)
- [x] Change edit command in DG & prerequisite details for each command (341, 342)
- [x] Edit command functionality bug (305)

closes #365 
closes #364 
closes #360 
closes #339 
closes #337 
closes #336 
closes #335
closes #334 
closes #327 
closes #312 
closes #308 
closes #288 
closes #286 
closes #341 
closes #342
closes #305

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -646,74 +646,87 @@ testers are expected to do more *exploratory* testing.
...
-    2. Test case: `addStu n/John Doe s/A1234567X t/T1001`<br>
+    1. Test case: `addStu n/John Doe s/A1234567X c/T1001`<br>
        Expected: A new student named "John Doe" with student ID "A1234567X" is added to tutorial "T1001". Confirmation message is displayed with the student's details.
 
 2. Adding a student with missing compulsory fields
 
-    1. Test case: `addStu n/John Doe t/T1001`<br>
-       Expected: Error message indicating that the student ID is missing.
+    1. Test case: `addStu n/John Doe c/T1001`<br>
+       Expected: Error message indicating that invalid command format will output.
```
Comment (Fui03) [inline docs/DeveloperGuide.md:681]: Good to see that expected error message is updated.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -372,6 +375,9 @@ Command detail & constraints
 - Existing values will be updated to the input values.
 - When a student’s tutorial class is updated, their attendance record will remain intact, as attendance is tracked at the module level, not by individual tutorial groups. This ensures that any classes the student attended before the change are still recorded accurately.
 
+Note:
+- Execute `edit` command after `find` command will edit the student data based on the filtered list after `find` command
```
Comment (Fui03) [inline docs/UserGuide.md:379]: Good to see the clarification on find command and edit command.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -188,8 +188,13 @@ First-time users are strongly encouraged to read the Getting Started section bef
    Some example commands you can try:
 
 * `list` : Lists all contacts.
+<<<<<<< HEAD
 * `addTut c/CS2103T id/T1001` : Adds a tutorial of name `CS2103T` and tutorial id `T1001`.
 * `addStu n/Samson s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id `T1001`.
+=======
+* `addTut c/CS2103T c/T1001` : Adds a tutorial of name `CS2103T` and tutorial id `T1001`.
+* `add n/Samson  s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id `T1001`.
+>>>>>>> fe7a195039dff2f88c66801eb4ee09b2f6ce3558
```
Comment (waihongteh) [inline docs/UserGuide.md:197]: Oops, seems like you forgot to remove the tag for merge conflict

File: docs/UserGuide.md [hunk 4]
```diff
@@ -188,13 +188,8 @@ First-time users are strongly encouraged to read the Getting Started section bef
...
 
 * `list` : Lists all contacts.
-<<<<<<< HEAD
-* `addTut c/CS2103T id/T1001` : Adds a tutorial of name `CS2103T` and tutorial id `T1001`.
-* `addStu n/Samson s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id `T1001`.
-=======
 * `addTut c/CS2103T c/T1001` : Adds a tutorial of name `CS2103T` and tutorial id `T1001`.
-* `add n/Samson  s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id `T1001`.
->>>>>>> fe7a195039dff2f88c66801eb4ee09b2f6ce3558
+* `addStu n/Samson s/A1234567X c/T1001` : Adds a student to the application and assigns him to tutorial with tutorial id `T1001`.
```
Comment (waihongteh) [inline docs/UserGuide.md:192]: addTut is removed? But adding a student with tutorial will require tutorial class to be created, perhaps you need to add it back in

Comment (waihongteh) [inline docs/UserGuide.md:192]: LGTM!

**Review by waihongteh — APPROVED** [review]

**Pull request merged**